### ISSUE TITLE
center loss on prelogits vs bottleneck

### ISSUE THREAD

**Issue by avasbr** [davidsandberg/facenet#262]
Is there a theoretical reason why center loss is computed on the prelogits layer and not on the embedding (bottleneck) layer? if the embeddings are what would eventually be used for clustering, recognition, etc, shouldn't center loss be applied to that? Apologies if this question has been asked already!

**Comment by davidsandberg** [comment]
Hi @avasbr,
Yes, that's a very relevant question. In the center loss paper they claim that the tendency to keep features closer to the centers will ripple back to also affect previous layers. But it would definitely be interesting to investigate.

**Comment by davidsandberg** [comment]
Hi again,
I moved the center loss to the bottleneck layer (mainly because I did some restructuring and then it got simpler) and the performance is unaffected. I take that it doesn't matter that much which layer the center loss is applied on, at least not for this model.

**Comment by avasbr** [comment]
ah, interesting - I've been taking some time to rewrite the nn4_small_v1 architecture in slim and just finished that up. I'm interested in trying to get a small deep model that could potentially fit on a mobile device  On that note, is there a way to get the inception resent models down in size? it appears to be around 115mb or so, whereas the nn4 model is about 20-25mb..
